app/controllers/schools/transferring_participants_controller.rb

```diff
@@ -214,9 +218,21 @@ def send_notification_emails!(new_induction_record, was_withdrawn_participant)
       )
     end
 
-    def set_cohort_to_transfer
-      cohort = participant_profile && !transferable_to_cohort_chosen? ? participant_cohort : active_cohort
-      set_school_cohort(cohort:)
+    def check_school_cohort
+      if participant_profile && participant_cohort != active_cohort
+        @school_cohort = @school.school_cohorts.find_by(cohort: participant_cohort)
+        if @school_cohort.present? && @school_cohort.full_induction_programme?
+          set_school_cohort(cohort: participant_cohort)
+        elsif action_name.to_sym != :need_training_setups
```

Comment:
Should this be `:need_training_setup`?

Reply:
Good spot, I noticed that just now, it doesn't need to be there actually as I put a condition on the before action